### PR TITLE
Arduino Build Support

### PR DESCRIPTION
**Pull request by ksedgwic**
Added Arduino library.properties metadata file.
Also needed some conditional non-compilation of test code which clashes in arduino env.
Needed to include <stdint.h>.

### CODE REVIEW

**Review by ChristopherA — DISMISSED** [review]
make check failed with errors.

```
make check
cd src && /Applications/Xcode.app/Contents/Developer/usr/bin/make check
cc -g -O0   -c -o encoding.o encoding.c
cc -g -O0   -c -o encrypt.o encrypt.c
cc -g -O0   -c -o mnemonics.o mnemonics.c
In file included from mnemonics.c:14:
/usr/local/include/bc-shamir/bc-shamir.h:13:10: fatal error: 'shamir.h' file not found
#include "shamir.h"
         ^~~~~~~~~~
1 error generated.
make[1]: *** [mnemonics.o] Error 1
make: *** [check] Error 2
```

**Comment by ksedgwic** [comment]
Does that work on master?  I don't think anything I've changed would cause that ...

**Comment by ChristopherA** [comment]
@wolfmcnally can you confirm master?

**Comment by ksedgwic** [comment]
Works on arduino.

I left the #include<stdint.h> in the test files too, seems like a good idea even though arduino no longer sees them ...

File: src/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# If we are using symbolic links for dependencies don't check the links in.
```
Comment (wolfmcnally) [inline src/.gitignore:1]: Please mention Arduino here. People who have no idea why these got added shouldn't have to track that info down.

File: src/encoding.c [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #include "wordlist-english.h"
 #include "util.h"
 
+#include <stdint.h>
```
Comment (wolfmcnally) [inline src/encoding.c:12]: You mentioned these <stdint.h> includes are no longer necessary for Arduino? If so, let's remove them.

**Review by wolfmcnally — DISMISSED** [review]
See comments.

File: src/.gitignore (hunk 1 above)
Comment (ksedgwic) [reply src/.gitignore:1]: I disagree, I considered that and other people may have to do this too.  *Any* embedded environment will need stuff like this.  Besides, you *should* be including stdint.h.

File: src/encoding.c (hunk 2 above)
Comment (ksedgwic) [reply src/encoding.c:12]: They *should* be there.

**Review by wolfmcnally — APPROVED** [review]
Temporary changes accepted with reservations.

**Pull request merged**